swap/protocol.go

```diff
@@ -121,8 +121,7 @@ func (s *Swap) run(p *p2p.Peer, rw p2p.MsgReadWriter) error {
 		return err
 	}
 
-	swapPeer := NewPeer(protoPeer, s, beneficiary, response.ContractAddress)
-	s.addPeer(swapPeer)
+	swapPeer := s.newPeer(protoPeer, beneficiary, response.ContractAddress)
```

Comment:
I would probably do it the other way around; usually `new` denotes just a construction, not additional work done. So I would call here `addPeer` but in `addPeer` call `NewPeer`

Reply:
done